### ISSUE TITLE
Giving Heading to Legends.

### ISSUE THREAD

**Issue by Ishansinghlamba** [recharts/recharts#2904]
I want to add heading to the legends. [
<img width="176" alt="Screenshot 2022-07-18 at 2 19 07 AM" src="https://user-images.githubusercontent.com/82999629/179424291-c9bdd866-1248-4e81-b5a7-797db13c2257.png">
](url)

**Comment by TooMuchLagWillKillYou** [comment]
You can render a custom Legend using the **content** prop in the **Legend** component. You can pass it a function (or a component, read the docs for more info) which will render the Legend.

This is an example: 
`  const CustomLegend = (props: any) => {
    const { payload } = props;

    return (
      <>
        <p style={{fontSize: "2rem", fontWeight: 800}}>Scale</p>
        <ul>
          {payload.map((entry: any, index: number) => (
            <li key={`item-${index}`}>{entry.value}</li>
          ))}
        </ul>
      </>
    );
  };` 

You can find the full chart here: https://codesandbox.io/s/customized-legend-event-forked-b32o5t

**Comment by ckifer** [comment]
This issue has an answer, but providing the ability to give a legend header may be a nice idea in general.

Labeling a feature request for now to come back to eventually.

**Comment by ckifer** [comment]
@bhatt40 yeah, `DefaultLegendContent` is exported now - https://github.com/recharts/recharts/blob/master/src/index.ts#L12

**Comment by bhatt40** [comment]
@ckifer Got it. Thanks.

**Comment by PavelVanecek** [comment]
Custom content is the answer.
